bundles/org.openhab.binding.visualcrossing/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,128 @@
+			</channel-group>
+			<channel-group id="day13" typeId="day-channel-group">
+				<label>Day T+13</label>
+			</channel-group>
+			<channel-group id="day14" typeId="day-channel-group">
+				<label>Day T+14</label>
+			</channel-group>
+			<channel-group id="day15" typeId="day-channel-group">
+				<label>Day T+15</label>
+			</channel-group>
+		</channel-groups>
+
+		<config-description>
+			<parameter name="apiKey" type="text" required="true">
+				<context>password</context>
+				<label>API Key</label>
+				<description>API Key to connect to the cloud.</description>
+			</parameter>
+			<parameter name="location" type="text">
+				<label>Location</label>
+				<description>Is the address, partial address or latitude,longitude location for which to retrieve weather data. You
+					can also use US ZIP Codes. If you would like to submit multiple locations in the same request, consider our
+					Multiple Location Timeline Weather API.</description>
+			</parameter>
+			<parameter name="lang" type="text">
+				<label>Language</label>
+				<description>Sets the language of the translatable parts of the output such as the conditions field.</description>
+				<options>
+					<option value="ar">Arabic</option>
+					<option value="bg">Bulgarian</option>
+					<option value="cs">Czech</option>
+					<option value="da">Danish</option>
+					<option value="de">German</option>
+					<option value="el">Greek</option>
+					<option value="en">English</option>
+					<option value="es">Spanish</option>
+					<option value="fa">Persian</option>
+					<option value="fi">Farsi</option>
+					<option value="fr">French</option>
+					<option value="he">Hebrew</option>
+					<option value="hu">Hungarian</option>
+					<option value="it">Italian</option>
+					<option value="ja">Japanese</option>
+					<option value="ko">Korean</option>
+					<option value="nl">Dutch</option>
+					<option value="pl">Polish</option>
+					<option value="pt">Portuguese</option>
+					<option value="ru">Russian</option>
+					<option value="sk">Slovakian</option>
+					<option value="sr">Serbian</option>
+					<option value="sv">Swedish</option>
+					<option value="tr">Turkish</option>
+					<option value="uk">Ukrainian</option>
+					<option value="vi">Vietnamese</option>
+					<option value="zh">Chinese</option>
+				</options>
+			</parameter>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>Hostname</label>
+				<description>Hostname or IP address of the server</description>
+				<advanced>true</advanced>
+				<default>https://weather.visualcrossing.com</default>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="1">
+				<label>Refresh Interval</label>
+				<description>Interval the device is polled in sec (default 1h).</description>
+				<default>3600</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="httpRetries" type="integer" min="0">
```

Comment:
```suggestion
			<parameter name="httpRetries" type="integer" min="1">
```
The `maxRetries` in the `RetryHttpClient` class throws an excpetion if it has 0 retries.

Reply:
But if you pass 0 then no retry will be applied. check handler

Reply:
Ah, if it is `0` tha other ApiClient is called, i see. In that case i would even stronger advice to integratei the httpclient and retryhttpclient. anyway, this comment is adressed. Thanks for pointing me at the correct check.

Reply:
But why should I integrate them? I've used decorator pattern to have/not have retries. ApiClient (old HttpClient) do not need to know anything about retries (or no retries)

Reply:
Pure simplification.

Reply:
It will be more complicated :D

Look at this joined code:

```java
private @Nullable String execute(Request request, @Nullable Header[] headers, String url)
        throws VisualCrossingApiException, VisualCrossingAuthException, VisualCrossingRateException {
    for (int i = 0; i < 10; i++) {
        try {
            try {
                if (headers != null) {
                    for (var header : headers) {
                        if (header == null) {
                            continue;
                        }
                        for (var value : header.values()) {
                            request.header(header.name(), value);
                        }
                    }
                }
                request.timeout(TIMEOUT, SECONDS);
                request.idleTimeout(IDLE_TIMEOUT, SECONDS);
                var response = request.send();
                var status = response.getStatus();
                if (status < 200 || status >= 399) {
                    throw new HttpVisualCrossingApiException(status, response.getReason());
                }
                return response.getContentAsString();
            } catch (RuntimeException | TimeoutException | ExecutionException | InterruptedException ex) {
                Throwable cause = ex;
                while (cause != null) {
                    if (cause instanceof HttpResponseException hte) {
                        var response = hte.getResponse();
                        int status = response.getStatus();
                        if (status == 401) {
                            throw new VisualCrossingAuthException();
                        }
                        if (status == 429) {
                            throw new VisualCrossingRateException();
                        }
                        throw new HttpVisualCrossingApiException(response.getStatus(), response.getReason(), hte);
                    }
                    cause = cause.getCause();
                }
                throw new VisualCrossingApiException("Error while executing request to " + url, ex);
            }
        } catch (VisualCrossingApiException ex) {
            if (i < 10 - 1) {
                logger.debug("Error while calling GET {}. Retrying {}/{}...", url, i + 1, 10, ex);
            } else {
                throw ex;
            }
        }
    }
    throw new IllegalStateException();
}
```